### ISSUE TITLE
Fix race condition in S3 delete expired tests

### ISSUE THREAD

**Issue by Murderlon** [tus/tus-node-server#533]
Regularly seeing a race condition occurring when two CI runs happen around the same time, which happens often if you have multiple PRs or a PR and a new commit in main. 

Details: https://github.com/tus/tus-node-server/actions/runs/7206763772/job/19632235153

**Comment by fenos** [comment]
Yes, this is a tricky one to fix, since they both try to call `s3.deleteExpired()` resulting in deleting each other files and the assertion then fails.

maybe we can create a new bucket with a random name during the tests and delete it right after
